docs/objects.md

```diff
@@ -4,63 +4,84 @@ title: Objects
+
+```rust
+let obj: Handle<JsObject> = cx.empty_object();
+```
 
 ## Getting Properties
 
-`.get()` is used to get a property of a JS object at runtime:
+The [`Object::get()`](https://docs.rs/neon/latest/neon/object/trait.Object.html#method.get) method accesses a property of an object at runtime:
 
 ```rust
-// --snip--
-let js_object = JsObject::new(&mut cx);
-js_object
-    .get(&mut cx, "myProperty")?
-    .downcast::<JsFunction>()
-    .or_throw(&mut cx)?;
-// --snip--
+// Create an empty object:
+let obj: Handle<JsObject> = cx.empty_object();
+
+// Get the `toString` property of the object:
+let prop: Handle<JsValue> = obj.get(&mut cx, "toString")?;
 ```
 
-`.downcast()` will attempt to cast the property to a `JsFunction`. `.or_throw()` will error if downcasting the propety is not possible.
+Notice that this example extracts the `toString` property from an empty object, which will typically be _inherited_ from the object's [prototype chain](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain).
 
 ## Setting Properties
 
-`.set()` requires a `FunctionContext`, the name of the property you want to set, and the value you want to set the property to:
+The [`Object::set()`](https://docs.rs/neon/latest/neon/object/trait.Object.html#method.set) method sets a property of an object at runtime:
 
 ```rust
-let js_object = JsObject::new(&mut cx);
-let js_string = cx.string("foobar");
+let obj = cx.empty_object();
+let age = cx.number(35);
 
-js_object.set(&mut cx, "myProperty", js_string)?;
+obj.set(&mut cx, "age", age)?;
 ```
 
-## Mapping a `struct` to a `JsObject`
+## Converting Rust data to JavaScript
 
-Here is a simple example of converting a Rust `struct` to a JS `Object` using `JsObject`. We first start by defining the `struct`:
+Here is a simple example of converting a Rust `struct` to a JavaScript object. First, let's define a Rust type describing, say, about a book in a library catalog:
 
 ```rust
-struct Foo {
-    pub bar: u64,
-    pub baz: String
+struct Book {
+    pub title: String,
+    pub author: String,
+    pub year: u64,
 }
 ```
-And then define a function which creates an instance of the `Foo` struct
+
+To copy a `Book` into a JavaScript object, we'll define a conversion function. Just for fun and to make it a little more idiomatically Rusty, we'll define it as a method of the `Book` type:
+
 ```rust
-fn convert_struct_to_js_object(mut cx: FunctionContext) -> JsResult<JsObject> {
-    let foo = Foo {
-        bar: 1234,
-        baz: "baz".to_string()
-    };
-    let object = JsObject::new(&mut cx);
-    let js_string = cx.string(&foo.baz);
-    let js_number = cx.number(foo.bar as f64);
-    object.set(&mut cx, "myStringProperty", js_string).unwrap();
-    object.set(&mut cx, "myNumberProperty", js_number).unwrap();
-    Ok(object)
+impl Book {
+    fn to_object(&self, mut cx: FunctionContext) -> JsResult<JsObject> {
```

Comment:
This is a nice terse example, but a more realistic one might borrow `Context` instead of taking ownership.

Also, I don't think the lifetime of the `JsResult` here can be ellided because the borrow of `self` would be inferred, which is incorrect.

Reply:
Thanks for pushing me on this, I think I have some nice improvements. I'll push them in a sec…